byteparse/shared/src/main/scala/byteparse/classparse/CodeParser.scala

```diff
@@ -0,0 +1,514 @@
+package byteparse
+package classparse
+
+import fastparse.allByte._
+
+object CodeParser {
+  sealed abstract class OpCode
+
+  case class AALoad() extends OpCode
```

Comment:
All the `case class`s which take no arguments we should make into `case object`s, to save allocations and improve perf across the board